### PR TITLE
Enable openmp parallelization for nms kernel

### PR DESCRIPTION
**Pull request by leslie-fang-intel**
In this PR:
- Add the OpenMP support to NMS CPU Kernel to enable the parallel calculation. 

- The MaskRCNN NMS kernel(https://github.com/facebookresearch/maskrcnn-benchmark/blob/main/maskrcnn_benchmark/csrc/cpu/nms_cpu.cpp#L5-L65) has offset of 1 when doing the calculation. Add the offset parameter(by default zero) to torch vision NMS kernel, then the MaskRCNN NMS kernel can share the implementation of Torchvision NMS Kernel.

### CODE REVIEW

**Review by datumbox — CHANGES_REQUESTED** [review]
@leslie-fang-intel Thanks a lot for your contribution.

I think it's worth splitting this PR to separate so that we don't mix the openmp parallelization with the introduction of bias. The parallelization work looks OK to me. I'm not sure that the change on the bias should be added though. My understanding is that it was not included intentionally as part of a simplification process. Also the introduction of the `bias` parameter on CPU only makes for a problematic API.

@fmassa do you have any thoughts on the above?

**Comment by vfdev-5** [comment]
I do not if this issue is related https://github.com/pytorch/vision/issues/2783 but previously we had a problem to make openmp available for torchvision...

**Comment by leslie-fang-intel** [comment]
Thanks for the suggestions. I will collect the benchmark number w and wo the parallelization.

**Comment by leslie-fang-intel** [comment]
Hi @datumbox @fmassa Here is the test case which I plan to collect performance with.
I suppose we will do `nms` for each batch and each class. So I will change the parameter of `batchsize`, `num_boxes`, `class_number` for each set of test and get the performance number with each `OMP_NUM_THREADS`.

Does this test case looks reasonable to you?
```
import torch
from torchvision import models, ops
import time

if __name__ == "__main__":
    batchsize=16
    num_boxes = 15130
    class_number = 81 # For coco dataset
    iou_threshold = 0.9

    boxes = torch.cat((torch.rand(num_boxes, 2), torch.rand(num_boxes, 2) + 10), dim=1) # 15130 * 4
    assert max(boxes[:, 0]) < min(boxes[:, 2])  # x1 < x2
    assert max(boxes[:, 1]) < min(boxes[:, 3])  # y1 < y2

    scores = torch.rand(num_boxes, class_number) # 15130 * 81

    steps = 10
    start = time.time()
    for _ in range(10):
        for batch_id in range(batchsize):
            for class_id in range(class_number):
                ops.boxes.nms(boxes[:, :], scores[:, class_id], iou_threshold)
    print("Toal time: {}".format(time.time() - start))

```

**Comment by datumbox** [comment]
@leslie-fang-intel Here is a [test](https://github.com/pytorch/vision/blob/11bd2eaa6d6976129836b329b01d1300babddcc9/test/test_onnx.py#L93) that does something similar. I would also fix the seeds for all tests to avoid differences related to the data. For example what you could do is generate different data per iteration by setting the seed equal to iteration number so that you get different results per iteration but same across experiments. Finally I would recommend doing a few warm up iters prior start measuring to ensure you get stable and accurate metrics.

**Comment by fmassa** [comment]
Hi @leslie-fang-intel 

I'd run the benchmark for a number of different boxes (not only 15130). Indeed, for a vanilla Faster R-CNN with FPN, the NMS kernel will see at most 2000 boxes, so it would be good to have a range like `[300, 1000, 2000, 5000, 10000]` for example

**Comment by leslie-fang-intel** [comment]
Will create new PR with parallelization optimization only as first step.
